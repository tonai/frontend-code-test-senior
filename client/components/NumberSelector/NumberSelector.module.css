.root {
  display: inline-block;
}

.label {
  text-align: center;
  font-size: 12px;
  padding: 6px 0;
}

.row {
  display: flex;
  align-items: center;
}

.value {
  display: inline-block;
  min-width: 32px;
  text-align: center;
  padding: 0 6px;
  font-weight: bold;
  font-size: 22px;
  line-height: 22px;
}
