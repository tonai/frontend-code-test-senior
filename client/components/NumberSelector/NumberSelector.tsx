import Button from "../Button/Button";

import css from "./NumberSelector.module.css";

export interface INumberSelectorProps {
  label: string;
  onChange?: (value: number) => void;
  maxValue?: number;
  minValue?: number;
  value: number;
}

export default function NumberSelector(props: INumberSelectorProps) {
  const { label, maxValue, minValue, onChange, value } = props;

  function handleDecrement() {
    onChange?.(value - 1);
  }

  function handleIncrement() {
    onChange?.(value + 1);
  }

  return (
    <div className={css.root}>
      <div className={css.label}>{label}</div>
      <div className={css.row}>
        <Button disabled={value <= minValue} onClick={handleDecrement} small>
          -
        </Button>
        <span className={css.value}>{value}</span>
        <Button disabled={value >= maxValue} onClick={handleIncrement} small>
          +
        </Button>
      </div>
    </div>
  );
}
