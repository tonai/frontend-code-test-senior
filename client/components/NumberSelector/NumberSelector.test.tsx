import { render } from "@testing-library/react";

import NumberSelector from "./NumberSelector";

describe("Layout component", () => {
  it("matches snapshot", () => {
    const { container } = render(<NumberSelector label="Qty" value={42} />);
    expect(container).toMatchSnapshot();
  });
});
