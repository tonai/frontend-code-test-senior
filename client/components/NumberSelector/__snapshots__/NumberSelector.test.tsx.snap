// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="label"
    >
      Qty
    </div>
    <div
      class="row"
    >
      <button
        class="button buttonSmall"
        type="button"
      >
        -
      </button>
      <span
        class="value"
        title="Current quantity"
      >
        42
      </span>
      <button
        class="button buttonSmall"
        type="button"
      >
        +
      </button>
    </div>
  </div>
</div>
`;
